antarest/study/storage/abstract_storage_service.py

```diff
@@ -337,6 +337,50 @@ def export_study(
             )
         return target
 
+    def _export_output(
+        self, metadata: T, output_id: str, target: Path
+    ) -> Path:
+        path_output = Path(metadata.path) / "output" / output_id
+        start_time = time.time()
+        with ZipFile(target, "w", ZIP_DEFLATED) as zipf:
+            current_dir = os.getcwd()
+            os.chdir(path_output)
+
+            for path in glob.glob("**", recursive=True):
+                zipf.write(path, path)
+
+            zipf.close()
+
+            os.chdir(current_dir)
+        stop_time = time.time()
+        duration = "{:.3f}".format(stop_time - start_time)
+        logger.info(
+            f"Output {output_id} from study {metadata.path} exported in {duration}s"
+        )
+        _, study = self.study_factory.create_from_fs(
+            target, "", use_cache=False
+        )
+        return target
+
+    @abstractmethod
+    def export_output(
+        self,
+        metadata: T,
+        output_id: str,
+        target: Path,
+    ) -> Path:
+        """
+        Export and compresse study inside zip
+        Args:
+            metadata: study
+            output_id: output id
+            target: path of the file to export to
+
+        Returns: zip file with study files compressed inside
+
+        """
+        raise NotImplementedError()
```

Comment:
it's the same method for both variant and raw studies, so you can put all the code here